internal/manifests/manifests.go

```diff
@@ -450,6 +477,24 @@ func (m *Manifests) validateUserSuppliedManifest(ctx context.Context, clusterID
 	return nil
 }
 
+// isValidYaml checks if one or more (in case of multi-doc yaml) yaml documents are valid
```

Comment:
Might be better to say that this checks if all yaml docs present in manifestContent are valid, that this may be more than one document in the case of multi-doc yaml

```
// isValidYaml checks if all yaml documents are valid, in the case of multi-doc yaml this may be more than one document.
```